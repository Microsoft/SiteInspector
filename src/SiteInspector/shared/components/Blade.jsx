import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const Blade = ({ visible, title, closeHandler, displayUI, isLargerWidth }) => {
  const BladeClassNames = () => classNames({
    visible,
    'larger-width': isLargerWidth,
  });

  return (
    <div className={`si-blade flex-column ${BladeClassNames()}`}>
      <div className="si-blade-header">
        <div className="si-blade-header-title">{title}</div>
        <a href="javascript:void(0)" className="si-close-panel-icon c-glyph glyph-clear" onClick={closeHandler} />
      </div>
      <div className="si-blade-content flex-column">
        {visible && displayUI}
      </div>
    </div>
  );
};

Blade.propTypes = {
  visible: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  closeHandler: PropTypes.func.isRequired,
  displayUI: PropTypes.element.isRequired,
  isLargerWidth: PropTypes.bool,
};

Blade.defaultProps = {
  isLargerWidth: false,
};

export default Blade;
