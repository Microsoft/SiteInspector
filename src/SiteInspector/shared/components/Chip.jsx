import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const Chip = ({ extraClassNames, glyph, title, href, target }) => (
  <div className={classNames('si-chip', extraClassNames)}>
    <a className="urlpad" href={href} target={target} aria-label={title}>
      {title}
    </a>
    <a href={href} target={target} aria-label="Redirect">
      <span className={glyph} />
    </a>
  </div>
);

Chip.propTypes = {
  extraClassNames: PropTypes.string,
  glyph: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  target: PropTypes.string.isRequired,
};

Chip.defaultProps = {
  extraClassNames: '',
};

export default Chip;
