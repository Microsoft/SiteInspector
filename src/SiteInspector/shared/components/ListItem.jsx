import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ListItem = ({ content, active }) => {
  const listItemClasses = () => classNames({
    active,
    'si-list-item': true,
  });

  return (
    <div className={listItemClasses()}>
      {content}
    </div>
  );
};

ListItem.propTypes = {
  content: PropTypes.shape().isRequired,
  active: PropTypes.bool,
};

ListItem.defaultProps = {
  active: false,
};

export default ListItem;
