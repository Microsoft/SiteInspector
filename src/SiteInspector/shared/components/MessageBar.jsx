import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { MessageBar as OfficeMessageBar } from 'office-ui-fabric-react/lib/MessageBar';

const MessageBar = ({ messages, removeMessage }) => (
  <div className="si-messages">
    {messages.map(message => (
      <OfficeMessageBar
        messageBarType={message.type
          ? message.type
          : 0}
        className={classnames({ collapsing: message.isCollapsing })}
        dismissButtonAriaLabel="Close"
        onDismiss={function () { removeMessage(message.key); }}
        key={message.key}
      >
        {message.text}
      </OfficeMessageBar>))}
  </div>
  );

MessageBar.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    isCollapsing: PropTypes.bool,
    key: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    type: PropTypes.number,
    timeout: PropTypes.number,
  })),
  removeMessage: PropTypes.func.isRequired,
};

MessageBar.defaultProps = {
  messages: {},
};

export default MessageBar;
