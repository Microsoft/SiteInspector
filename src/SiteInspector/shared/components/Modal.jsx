import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';

const Modal = ({ visible, title, closeHandler, displayUI, contentOverride }) => (
  <ReactModal
    isOpen={visible}
    onRequestClose={closeHandler}
    style={contentOverride}
    contentLabel={title}
    overlayClassName="si-modal-overlay"
    className="si-modal-dialog"
    shouldCloseOnOverlayClick
  >
    <div className="si-modal-content">
      <div className="si-modal-header">
        <div className="si-modal-header-title">{title}</div>
        <a href="javascript:void(0)" className="si-close-panel-icon c-glyph glyph-clear" onClick={closeHandler} />
      </div>
      <div className="si-modal-body">
        {displayUI}
      </div>
    </div>
  </ReactModal>
);

Modal.propTypes = {
  visible: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  closeHandler: PropTypes.func.isRequired,
  displayUI: PropTypes.node.isRequired,
  contentOverride: PropTypes.shape(),
};

Modal.defaultProps = {
  contentOverride: undefined,
};

export default Modal;
