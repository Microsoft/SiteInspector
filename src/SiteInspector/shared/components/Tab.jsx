/* eslint jsx-a11y/aria-props:"off" */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const SITab = ({ isActive, tab, setViewer }) => (
  <a
    className={`si-tab c-select-button ${classNames({ active: isActive })}`}
    aria-current={isActive}
    href="javascript:void(0)" onClick={() => setViewer(tab.id)}
  >{tab.label}</a>);

SITab.propTypes = {
  isActive: PropTypes.bool.isRequired,
  tab: PropTypes.shape({
    id: PropTypes.string.isRequired,
    label: PropTypes.node.isRequired,
  }).isRequired,
  setViewer: PropTypes.func.isRequired,
};

export default SITab;
