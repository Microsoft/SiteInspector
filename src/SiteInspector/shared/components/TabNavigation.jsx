import React from 'react';
import PropTypes from 'prop-types';
import Tab from './Tab';

const TabNavigation = ({ tabs, activeTab, setTab }) => {
  const makeTab = (key) => {
    const tab = tabs[key];
    return (<Tab
      key={key} isActive={activeTab === tab.id}
      tab={tab} setViewer={setTab}
    />);
  };

  const resolveactiveTab = () => tabs[activeTab].view;

  return (
    <div className="si-tab-navigation flex-column">
      <div className="si-nav">
        <div className="c-group" role="tablist">
          {
            Object.keys(tabs).map(makeTab)
          }
        </div>
      </div>
      <div className="si-tab-content flex-column">
        {resolveactiveTab()}
      </div>
    </div>
  );
};

TabNavigation.propTypes = {
  activeTab: PropTypes.string.isRequired,
  tabs: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    view: PropTypes.element.isRequired,
  })).isRequired,
  setTab: PropTypes.func.isRequired,
};

export default TabNavigation;
