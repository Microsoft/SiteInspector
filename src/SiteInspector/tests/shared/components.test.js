// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Blade from '../../shared/components/Blade';
import Chip from '../../shared/components/Chip';
import ListItem from '../../shared/components/ListItem';
import MessageBar from '../../shared/components/MessageBar';
import Modal from '../../shared/components/Modal';
import Tab from '../../shared/components/Tab';
import TabNavigation from '../../shared/components/TabNavigation';

configure({ adapter: new Adapter() });

describe('Shared components', () => {
  describe('Blade component', () => {
    let blade;

    beforeEach(() => {
      blade = shallow(<Blade
        title="test title"
        closeHandler={jest.fn()}
        displayUI={<div>hello world</div>}
        isLargerWidth={false}
      />);
    });

    it('should be created', () => {
      expect(blade.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(blade).toMatchSnapshot();
    });
  });

  describe('Chip component', () => {
    let chip;

    beforeEach(() => {
      chip = shallow(<Chip
        extraClassNames={null}
        glyph="glyph"
        title="test title"
        href="#"
        target="_blank"
      />);
    });

    it('should be created', () => {
      expect(chip.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(chip).toMatchSnapshot();
    });
  });

  describe('ListItem component', () => {
    let listItem;

    beforeEach(() => {
      listItem = shallow(<ListItem
        content={<div>test</div>}
      />);
    });

    it('should be created', () => {
      expect(listItem.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(listItem).toMatchSnapshot();
    });
  });

  describe('MessageBar component', () => {
    let msgBar;
    const testMsg = {
      isCollapsing: false,
      key: 'test',
      text: 'test message',
      type: 1,
      timeout: 10,
    };

    beforeEach(() => {
      msgBar = shallow(<MessageBar
        messages={[testMsg]}
        removeMessage={jest.fn()}
      />);
    });

    it('should be created', () => {
      expect(msgBar.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(msgBar).toMatchSnapshot();
    });
  });

  describe('Modal component', () => {
    let modal;

    beforeEach(() => {
      modal = shallow(<Modal
        title="test title"
        closeHandler={jest.fn()}
        displayUI={<div>test</div>}
        isLargerWidth={false}
      />);
    });

    it('should be created', () => {
      expect(modal.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(modal).toMatchSnapshot();
    });
  });

  describe('Tab component', () => {
    let tab;
    const tabData = {
      id: 'test',
      label: 'test label',
    };

    beforeEach(() => {
      tab = shallow(<Tab
        tab={tabData}
        setViewer={jest.fn()}
      />);
    });

    it('should be created', () => {
      expect(tab.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(tab).toMatchSnapshot();
    });
  });

  describe('TabNavigation component', () => {
    let tabNav;
    const tabData = {
      id: 'test',
      label: 'test label',
      view: <div>test</div>,
    };

    beforeEach(() => {
      tabNav = shallow(<TabNavigation
        tabs={{ test: tabData }}
        activeTab="test"
        setTab={jest.fn()}
      />);
    });

    it('should be created', () => {
      expect(tabNav.length).toEqual(1);
    });

    it('should be rendered properly', () => {
      expect(tabNav).toMatchSnapshot();
    });
  });
});
