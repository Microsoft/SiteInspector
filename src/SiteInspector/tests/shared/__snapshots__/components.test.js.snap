// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared components Blade component should be rendered properly 1`] = `
<div
  className="si-blade flex-column visible"
>
  <div
    className="si-blade-header"
  >
    <div
      className="si-blade-header-title"
    >
      test title
    </div>
    <a
      className="si-close-panel-icon c-glyph glyph-clear"
      href="javascript:void(0)"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="si-blade-content flex-column"
  >
    <div>
      hello world
    </div>
  </div>
</div>
`;

exports[`Shared components Chip component should be rendered properly 1`] = `
<div
  className="si-chip"
>
  <a
    aria-label="test title"
    className="urlpad"
    href="#"
    target="_blank"
  >
    test title
  </a>
  <a
    aria-label="Redirect"
    href="#"
    target="_blank"
  >
    <span
      className="glyph"
    />
  </a>
</div>
`;

exports[`Shared components ListItem component should be rendered properly 1`] = `
<div
  className="active si-list-item"
>
  <div>
    test
  </div>
</div>
`;

exports[`Shared components MessageBar component should be rendered properly 1`] = `
<div
  className="si-messages"
>
  <MessageBar
    className=""
    dismissButtonAriaLabel="Close"
    isMultiline={true}
    key="test"
    messageBarType={1}
    onDismiss={[Function]}
  >
    test message
  </MessageBar>
</div>
`;

exports[`Shared components Modal component should be rendered properly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="si-modal-dialog"
  closeTimeoutMS={0}
  contentLabel="test title"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="si-modal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <div
    className="si-modal-content"
  >
    <div
      className="si-modal-header"
    >
      <div
        className="si-modal-header-title"
      >
        test title
      </div>
      <a
        className="si-close-panel-icon c-glyph glyph-clear"
        href="javascript:void(0)"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="si-modal-body"
    >
      <div>
        test
      </div>
    </div>
  </div>
</Modal>
`;

exports[`Shared components Tab component should be rendered properly 1`] = `
<a
  aria-current={true}
  className="si-tab c-select-button active"
  href="javascript:void(0)"
  onClick={[Function]}
>
  test label
</a>
`;

exports[`Shared components TabNavigation component should be rendered properly 1`] = `
<div
  className="si-tab-navigation flex-column"
>
  <div
    className="si-nav"
  >
    <div
      className="c-group"
      role="tablist"
    >
      <SITab
        isActive={true}
        key="test"
        setViewer={[MockFunction]}
        tab={
          Object {
            "id": "test",
            "label": "test label",
            "view": <div>
              test
            </div>,
          }
        }
      />
    </div>
  </div>
  <div
    className="si-tab-content flex-column"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
